domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsExactlyEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,298 @@
+import org.oppia.android.app.model.Fraction
+import org.oppia.android.app.model.InteractionObject
+import org.oppia.android.domain.classify.RuleClassifier
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [FractionInputIsExactlyEqualToRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class FractionInputIsExactlyEqualToRuleClassifierProviderTest {
+  /**
+   * Store fractions in format:
+   *    <0 or 1 indicating whether it's negative>, <numerator>, <denominator>
+   *    e.g. -1/2 would be {1, 1, 2}
+   * Store whole numbers in format:
+   *    <0 or 1 indicating whether it's negative>, <whole number>
+   *    e.g. 1234 would be {0, 1234}
+   * Store mixed numbers in format:
+   *    <0 or 1 indicating whether it's negative>, <whole number>, <numerator>, <denominator>
+   *    e.g. -1234 1/2 would be {1, 1234, 1, 2}
+   */
+  private val NON_NEGATIVE_VALUE_0 = createNonNegativeInt(value = 0)
+  private val WHOLE_POS_1 = listOf(0, 123)
+  private val WHOLE_POS_2 = listOf(0, 321)
+  private val FRAC_POS_1 = listOf(0, 2, 4)
+  private val FRAC_POS_2 = listOf(0, 1, 2)
+  private val FRAC_POS_3 = listOf(0, 123, 1)
+  private val MIXED_POS_1 = listOf(0, 123, 1, 2)
+  private val MIXED_POS_2 = listOf(0, 123, 1, 3)
+  private val MIXED_NEG_1 = listOf(1, 123, 1, 2)
+
+  @Inject
+  internal lateinit var fractionInputIsExactlyEqualToRuleClassifierProvider:
+    FractionInputIsExactlyEqualToRuleClassifierProvider
+
+  private val isExactlyEqualClassifierProvider: RuleClassifier by lazy {
+    fractionInputIsExactlyEqualToRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Test
+  fun testAnswer_testFraction_bothWholeNumbers_bothValuesMatch() {
+    val inputs = mapOf("f" to createWholeNumberOnly(WHOLE_POS_1))
+
+    val matches =
+      isExactlyEqualClassifierProvider.matches(
+        answer = createWholeNumberOnly(WHOLE_POS_1),
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer_testFraction_bothWholeNumbers_bothValuesDoNotMatch() {
+    val inputs = mapOf("f" to createWholeNumberOnly(WHOLE_POS_1))
+
+    val matches =
+      isExactlyEqualClassifierProvider.matches(
+        answer = createWholeNumberOnly(WHOLE_POS_2),
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer_testFraction_bothFractions_bothValuesMatch() {
+    val inputs = mapOf("f" to createFractionOnly(FRAC_POS_1))
+
+    val matches =
+      isExactlyEqualClassifierProvider.matches(
+        answer = createFractionOnly(FRAC_POS_1),
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer_testFraction_bothFractions_oneFractionReduced_bothValuesDoNotMatch() {
```

Comment:
Related to a comment above, but it's tricky to verify the 'one fraction reduced' condition without slightly clearer constant names.